keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -97,10 +110,12 @@ class ApplicationService(
               )
             }?.let { artifactSummaryInEnvironment ->
               addStatefulConstraintSummaries(artifactSummaryInEnvironment, deliveryConfig, environment, version)
-              // TODO: add stateless constraint summaries
-            }?.also { artifactSummaryInEnvironment ->
-              artifactSummariesInEnvironments.add(artifactSummaryInEnvironment)
             }
+              ?.let { artifactSummaryInEnvironment ->
```

Comment:
You don't need the extra `.let` here -- you can just call `addStatelessConstraintSummaries` right below `addStatefulConstraintSummaries` above, where the `//TODO` was.

Reply:
actually I've tried it! but when doing so, the StatefulConstraint got disappeared!